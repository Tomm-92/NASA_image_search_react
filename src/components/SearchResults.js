import React from "react";
import "../styles/searchResults.css";

const SearchResults = ({ hasSearched, results }) => {
  if (!hasSearched) return null;
  if (!results.length) {
    return (
      <div>
        <h3> No results</h3>
        <p> Please try another search </p>
      </div>
    );
  }

  return (
    <div className="image-gallery">
      {results.map((image) => (
        <img
          className="return-images"
          src={image}
          alt="spaceImage"
          key={image}
        />
      ))}
    </div>
  );
};

export default SearchResults;
