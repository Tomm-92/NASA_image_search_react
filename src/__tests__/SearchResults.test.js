import React from "react";
import { render, screen } from "@testing-library/react";
import SearchResults from "../components/SearchResults";

describe("Search Results", () => {
  const validProps = {
    hasSearched: jest.fn(),
    results: [],
  };

  it("renders correctly", () => {
    const { asFragment } = render(
      <SearchResults
        results={validProps.results}
        hasSearched={validProps.hasSearched}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders no results message when there are no results", () => {
    render(<SearchResults results={[]} hasSearched={true} />);
    const noResultsMessage = screen.getByText("No results");
    const anotherSearchMessage = screen.getByText("Please try another search");

    expect(noResultsMessage).toBeInTheDocument();
    expect(anotherSearchMessage).toBeInTheDocument();
  });

  it("renders images when there are results to return", () => {
    const results = [
      "search-image1",
      "search-image2",
      "search-image3",
      "search-image4",
    ];
    render(<SearchResults results={results} hasSearched={true} />);
    const imageResults = screen.queryAllByAltText("spaceImage");
    expect(imageResults.length).toBe(4);
  });
});
