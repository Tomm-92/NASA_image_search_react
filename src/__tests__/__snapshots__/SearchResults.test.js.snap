// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Results renders correctly 1`] = `
<DocumentFragment>
  <div>
    <h3>
       No results
    </h3>
    <p>
       Please try another search 
    </p>
  </div>
</DocumentFragment>
`;
